google/cloud/bigquery_storage_v1/reader.py

```diff
@@ -0,0 +1,608 @@
+            ]):
+                The ReadRows stream to read.
+            client ( \
+                ~google.cloud.bigquery_storage_v1.gapic. \
+                    big_query_read_client.BigQueryReadClient \
+            ):
+                A GAPIC client used to reconnect to a ReadRows stream. This
+                must be the GAPIC client to avoid a circular dependency on
+                this class.
+            name (str):
+                Required. Stream ID from which rows are being read.
+            offset (int):
+                Required. Position in the stream to start
+                reading from. The offset requested must be less than the last
+                row read from ReadRows. Requesting a larger offset is
+                undefined.
+            read_rows_kwargs (dict):
+                Keyword arguments to use when reconnecting to a ReadRows
+                stream.
+
+        Returns:
+            Iterable[ \
+                ~google.cloud.bigquery_storage_v1.types.ReadRowsResponse \
+            ]:
+                A sequence of row messages.
+        """
+
+        # Make a copy of the read position so that we can update it without
+        # mutating the original input.
+        self._wrapped = wrapped
+        self._client = client
+        self._name = name
+        self._offset = offset
+        self._read_rows_kwargs = read_rows_kwargs
+
+    def __iter__(self):
+        """An iterable of messages.
+
+        Returns:
+            Iterable[ \
+                ~google.cloud.bigquery_storage_v1.types.ReadRowsResponse \
+            ]:
+                A sequence of row messages.
+        """
+
+        # Infinite loop to reconnect on reconnectable errors while processing
+        # the row stream.
+        while True:
+            try:
+                for message in self._wrapped:
+                    rowcount = message.row_count
+                    self._offset += rowcount
+                    yield message
+
+                return  # Made it through the whole stream.
+            except google.api_core.exceptions.InternalServerError as exc:
+                resumable_error = any(
+                    resumable_message in exc.message
+                    for resumable_message in _STREAM_RESUMPTION_INTERNAL_ERROR_MESSAGES
+                )
+                if not resumable_error:
+                    raise
+            except _STREAM_RESUMPTION_EXCEPTIONS:
+                # Transient error, so reconnect to the stream.
+                pass
+
+            self._reconnect()
+
+    def _reconnect(self):
+        """Reconnect to the ReadRows stream using the most recent offset."""
+        self._wrapped = self._client.read_rows(
+            self._name, self._offset, **self._read_rows_kwargs
+        )
+
+    def rows(self, read_session):
+        """Iterate over all rows in the stream.
+
+        This method requires the fastavro library in order to parse row
```

Comment:
Looks like I never updated this comment to account for arrow streams.

Reply:
updated.